globus_sdk/transfer/client.py

```diff
@@ -334,6 +334,67 @@ def submit_delete(self, data):
         """
         return self.post('/delete', data)
 
+    def make_submit_transfer_data(self, source_endpoint, dest_endpoint,
+                                  transfer_items, label, sync_level=None,
+                                  **kwargs):
```

Comment:
Regardless of whether we allow label to be None, if as @jaswilli says that the API will complain on the empty string, we should catch that in the SDK and report that as an error/exception rather than waiting for the API to return with an exception.